firmware/quadruna/VC/src/app/app_soc.c

```diff
@@ -0,0 +1,84 @@
+#include "app_soc.h"
+#include "io_currentSensing.h"
+#include "app_math.h"
+#include "app_canTx.h"
+#include "io_lowVoltageBattery.h"
+
+static SocStats stats;
+
+void app_soc_init(void)
+{
+    stats.prev_current_A = 0.0f;
+
+    stats.charge_c = app_soc_getChargeFromOcv();
```

Comment:
Ideally, this value would be stored in persistent memory (SD), as the voltage of the cells only mimic "Open-circuit" conditions after long periods of settling with no current draw (~1 hour).

Resets based on OCV are still required, but should be done so manually if possible.

Reply:
Will implement, currently looks like the SD card io interface isn't complete yet, will add todo note